### PR TITLE
Added tags

### PR DESCRIPTION
**Pull request by jeetbhalani15**
## Added Tag/Label Functionality.
- Added Tag component in Textarea & in Notes Card.
- Add custom tags, predefine tags, & Delete tags.
- Added tag page which contains all notes separated by tags category.
- Added chip component to show added Tags.

### CODE REVIEW

**Comment by Jay-Gadhiya** [comment]
- here is bug: when i delete chips, and edit note the chips f tags will appear again.

**Comment by Jay-Gadhiya** [comment]
- code looks good
- ui looks very good and input box is very great.
- just fix above mention bug.

**Review by Jay-Gadhiya — APPROVED** [review]

File: Gravity-Notes/src/Components/Archive-card/ArchiveCard.js [hunk 1]
```diff
@@ -22,13 +22,29 @@ export const ArchiveCard = ({ archive })=>{
...
+              &times;
+            </span>
+          </div>
+        ))}
+      </div>
         <div className={`cardicon ${archive.color}`}>
             <button className="card-action-btn"><MdMoreVert size={25} /></button>
             <button className="card-action-btn" onClick={() => deleteFromArchive(archive,authState,noteDispatch) }><MdDelete size={25} /></button>
             <button className="card-action-btn" onClick={() => restoreArchiveNote(archive,authState,noteDispatch) }><BiArchiveOut size={25} /></button>
             {/* <button onClick={() => editHandler(note)}><BiEditAlt size={25} /></button> */}
```
Comment (hrishikeshmane) [inline Gravity-Notes/src/Components/Archive-card/ArchiveCard.js:45]: remove commented code.

File: Gravity-Notes/src/Components/Notes-card/Notecard.js [hunk 2]
```diff
@@ -1,76 +1,280 @@
...
+            className="card-color-btn"
+          >
+            {" "}
+            <MdOutlineFormatColorReset size={5} />{" "}
+          </button>
+          <button
+            onClick={() => applyColorOnCard("red", note)}
+            className="card-color-btn red"
+          >
+            {" "}
```
Comment (hrishikeshmane) [inline Gravity-Notes/src/Components/Notes-card/Notecard.js:152]: remove formatted spaces.

File: Gravity-Notes/src/Components/Notes-card/Notecard.js [hunk 3]
```diff
@@ -1,76 +1,280 @@
...
+            <MdOutlineFormatColorReset size={5} visibility={"hidden"} />{" "}
+          </button>
+          <button
+            onClick={() => applyColorOnCard("orange", note)}
+            className="card-color-btn orange"
+          >
+            {" "}
```
Comment (hrishikeshmane) [inline Gravity-Notes/src/Components/Notes-card/Notecard.js:159]: same here.

File: Gravity-Notes/src/Components/Sidebar/Sidebar.js [hunk 4]
```diff
@@ -10,48 +10,74 @@ import { Archive } from "../../Pages/Archive/Archive-page";
...
+                </Link>
+                <span  className="tag">Tags</span><span className="badge">{note.tags.length}</span>
+
+            </li>
+            {/* {noteState.label.length > 0 && (
+                 <ul >
+                 {noteState.label.map((label,id)=>(<div><li key={id} >{label}</li></div>))}
+         </ul>
+            ) } */}
+
```
Comment (hrishikeshmane) [inline Gravity-Notes/src/Components/Sidebar/Sidebar.js:62]: remove this.

File: Gravity-Notes/src/Components/Sidebar/Sidebar.js [hunk 5]
```diff
@@ -10,48 +10,74 @@ import { Archive } from "../../Pages/Archive/Archive-page";
...
+            <li className="item">
                 <Link to={"/trash"}>
                 <button className="sidebar-btn" ><a href="#"><VscTrash size={30}/></a></button>
                 </Link>
                 <span  className="tag">Trash</span><span className="badge">{noteState.trashNotes.length}</span>
-            </div>
+            </li>
+            {/* <li className="label-box">
+                <input className="label-input" type="text" value={newLabel} onChange={(e)=> setNewLabel(e.target.value)} placeholder="Enter label"  /><IoIosAdd className="label-btn" onClick={()=> addNewLabelInList()} size={20}/>
+            </li> */}
```
Comment (hrishikeshmane) [inline Gravity-Notes/src/Components/Sidebar/Sidebar.js:78]: this aswell

File: Gravity-Notes/src/Components/Tags-card/TagCard.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    return(
+            <div className={`note ${tag.color}`}>
+            <div className="title-box">
+            <h1 className="title-txt">{tag.title}</h1>
+            <small>{tag.data}</small>
+            </div>
+            <p dangerouslySetInnerHTML={{ __html: tag.content}}/>
+            <div className={`cardicon ${tag.color}`}>
+                {/* <button className="card-action-btn" onClick={()=> deleteFromTrash(tag)} ><MdDelete size={25} /></button> */}
+                {/* <button className="card-action-btn" onClick={(e)=> restoreFromTrash(e,trash)}><FaTrashRestore size={25} /></button> */}
```
Comment (hrishikeshmane) [inline Gravity-Notes/src/Components/Tags-card/TagCard.js:18]: this too

File: Gravity-Notes/src/Pages/Tags-page/Tag-page.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+                 if(item.tags.find(tag => tag !== "work" && tag !== "personal" && tag !== "health" && tag !== "medication" && tag !== "fitness" && tag !== "study")){
+                    return  <TagCard key={item._id} tag={item} />
+                  }
+                })}
+              </div>
+            </section>
+          </div>
+            {/* {note.tags.map((item) => (
+              <TagCard key={item._id} tag={item} />
+            ))} */}
```
Comment (hrishikeshmane) [inline Gravity-Notes/src/Pages/Tags-page/Tag-page.js:113]: this too.

**Review by hrishikeshmane — APPROVED** [review]
Looks good other than formatted empty spaces and commented code.

**Pull request merged**